src/cli/src/cluster/install/tls.rs

```diff
@@ -0,0 +1,69 @@
+use std::path::PathBuf;
+use tracing::debug;
+use structopt::StructOpt;
+use fluvio::config::{TlsPolicy, TlsPaths};
+
+#[derive(Debug, StructOpt)]
+pub struct TlsOpt {
+    /// tls
+    #[structopt(long)]
+    pub tls: bool,
+
+    /// TLS: domain
+    #[structopt(long, required_if("tls", "true"))]
+    pub domain: Option<String>,
+
+    /// TLS: path to server certificate
+    #[structopt(long, required_if("tls", "true"), parse(from_os_str))]
+    pub server_cert: Option<PathBuf>,
+
+    /// TLS: path to server private key
+    #[structopt(long, required_if("tls", "true"), parse(from_os_str))]
+    pub server_key: Option<PathBuf>,
+
+    /// TLS: client cert
+    #[structopt(long, required_if("tls", "true"), parse(from_os_str))]
+    pub client_cert: Option<PathBuf>,
+
+    /// TLS: client key
+    #[structopt(long, required_if("tls", "true"), parse(from_os_str))]
+    pub client_key: Option<PathBuf>,
+
+    /// TLS: ca cert
+    #[structopt(long, required_if("tls", "true"), parse(from_os_str))]
+    pub ca_cert: Option<PathBuf>,
+}
+
+impl From<TlsOpt> for (TlsPolicy, TlsPolicy) {
```

Comment:
add unit test

Reply:
*sigh* looks like StructOpt has a weird problem where `required_if` does not work for a boolean field.
I.e. all of the `required_if("tls", "true")` calls are doing nothing at all :(

Reply:
I rolled back to custom validation